tests/integration/test_api.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import absolute_import, print_function
+import sys
+import os
+import uuid
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", ".."))
+
+from webserver.testing import ServerTestCase
+from flask import url_for
+import db.user
+import time
+import json
+
+# lifted from AcousticBrainz
+def is_valid_uuid(u):
+    try:
+        u = uuid.UUID(u)
+        return True
+    except ValueError:
+        return False
+
+class APITestCase(ServerTestCase):
+
+    def setUp(self):
+        self.user = db.user.get_or_create('testuserpleaseignore')
+
+    def test_get_listens(self):
+        """ Test to make sure that the api sends valid listens on get requests.
+        """
+        with open(os.path.join(os.getcwd(), 'testdata', 'valid_single.json'), 'r') as f:
+            payload = json.load(f)
+        payload['payload'][0]['listened_at'] = int(time.time())
+        response = self.send_data(payload)
+        self.assert200(response)
+        time.sleep(10)
+        url = url_for('api_v1.get_listens', user_name = self.user['musicbrainz_id'])
+        response = self.client.get(url, query_string = {'count': '1'})
+        self.assert200(response)
+        data = json.loads(response.data)
+        self.assertTrue('payload' in data)
+        data = data['payload']
+        # make sure user id is correct
+        self.assertTrue('user_id' in data)
+        self.assertEquals(data['user_id'], self.user['musicbrainz_id'])
```

Comment:
I normally don't have two tests for something like this - if `'user_id'` is missing from the data then the test is going to fail anyway. We may as well make the test a bit smaller and just have the `assertEquals`